src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/contact/ContactResource.kt

```diff
@@ -160,7 +159,8 @@ class ContactResource(private val contactService: ContactService) {
     ]
   )
   fun getContactByPrisonNumber(@PathVariable prisonNumber: String, authentication: Authentication): ContactResponse =
-    toContactResponse(contactService.getContactByPrisonNumber(authentication.name, prisonNumber))
+    contactService.getContactByPrisonNumber(authentication.name, prisonNumber)
+      ?: throw ResourceNotFoundException("Could not find a matching Contact [${authentication.name}, $prisonNumber]")
```

Comment:
As this exception is tightly coupled to the resource layer (in the ExceptionHandler) I moved this to the resource. (the tight coupling argument could also be applied to ContactRequest and ContactResponse but an anti-corruption layer seems over the top at this stage... if we get more consumers of the service layer then it might be worth revisiting)